### DISCUSSION TITLE
Extension should be able to clean up after uninstall.

### DISCUSSION THREAD

**Discussion by Boegie19** [directus/directus#20603] [Feature Requests]
### Summary

A way to undo any and all changes the extension made on uninstall

### Motivation

There needs to be a way for extension to undo any and all changes the extension made when uninstalled.

### Detailed Design

some how save the migration down and arbitrary JavaScript function in the database so that they can be ran after you uninstalled the exension. after they are both run both should also be cleaned up.

### Must Have:
- Remove records on extension uninstall
-  Remove DB tables on extension uninstall

### Should Have:
- A way to run arbitrary JavaScript so you can call 3rd party services and remove records.

### Drawbacks

- You don't have access to the dependencies of the extension (Meaning that if they use a 3rd party lib we can't call it)

### Alternatives

- only do this for uninstalls from the directus UI
  - pros:
    - You have access to the all installed dependencies of the extension
  - cons:
    - If uninstalled by removing the exension menualy the changes are not undone
  

### Adoption Strategy

flag inside of the migration exensions to be cloned into the db and use down on uninstall.
a function that if exists gets cloned into the db and run on uninstall.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
